File: packages/sdk-java/src/test/java/xyz/alphahuman/sdk/IntegrationTest.java
package xyz.alphahuman.sdk;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;

import java.util.Map;

import static org.junit.jupiter.api.Assertions.*;

@EnabledIfEnvironmentVariable(named = "ALPHAHUMAN_TOKEN", matches = ".+")
class IntegrationTest {

    @Test
    void insertRecallQueryDeleteLifecycle() throws InterruptedException {
        String token = System.getenv("ALPHAHUMAN_TOKEN");
        String namespace = "integration-test-java-" + System.nanoTime();

        try (AlphahumanMemoryClient client = new AlphahumanMemoryClient(token)) {

            // --- Insert ---
            long now = System.currentTimeMillis() / 1000;
            InsertMemoryResponse insertResp = client.insertMemory(
                    new InsertMemoryParams("test-key-1", "The capital of France is Paris.", namespace)
                            .setMetadata(Map.of("source", "integration-test"))
                            .setCreatedAt(now)
                            .setUpdatedAt(now));
            assertTrue(insertResp.isSuccess(), "insert should succeed");
            System.out.println("Insert status: " + insertResp.getStatus());

            // Give the backend time to index
            Thread.sleep(2000);

            // --- Recall ---
            RecallMemoryResponse recallResp = client.recallMemory(
                    new RecallMemoryParams().setNamespace(namespace));
            assertTrue(recallResp.isSuccess(), "recall should succeed");
            System.out.println("Recall cached=" + recallResp.isCached()
                    + " llmContext=" + (recallResp.getLlmContextMessage() != null));

            // --- Query ---
            QueryMemoryResponse queryResp = client.queryMemory(
                    new QueryMemoryParams("What is the capital of France?").setNamespace(namespace));
            assertTrue(queryResp.isSuccess(), "query should succeed");
            System.out.println("Query cached=" + queryResp.isCached()
                    + " llmContext=" + (queryResp.getLlmContextMessage() != null));

            // --- Delete ---
            DeleteMemoryResponse deleteResp = client.deleteMemory(
                    new DeleteMemoryParams().setNamespace(namespace));
            assertTrue(deleteResp.isSuccess(), "delete should succeed");
            System.out.println("Deleted nodes: " + deleteResp.getNodesDeleted());

            // Give the backend time to process deletion
            Thread.sleep(1000);

            // --- Verify deletion ---
            RecallMemoryResponse verifyResp = client.recallMemory(
                    new RecallMemoryParams().setNamespace(namespace));
            assertTrue(verifyResp.isSuccess());
            System.out.println("After delete: context=" + verifyResp.getContext()
                    + " llmCtx=" + verifyResp.getLlmContextMessage());
        }
    }
}

File: packages/sdk-csharp/test/Alphahuman.Sdk.Tests/IntegrationTest.cs
using Xunit;

namespace Alphahuman.Sdk.Tests;

[Trait("Category", "Integration")]
public class IntegrationTest
{
    [Fact]
    public async Task InsertRecallQueryDeleteLifecycle()
    {
        var token = Environment.GetEnvironmentVariable("ALPHAHUMAN_TOKEN");
        if (string.IsNullOrEmpty(token))
        {
            // Skip: ALPHAHUMAN_TOKEN not set
            return;
        }

        var ns = $"integration-test-csharp-{DateTimeOffset.UtcNow.ToUnixTimeMilliseconds()}";
        using var client = new AlphahumanMemoryClient(token);

        // ── Insert ──
        var nowSeconds = DateTimeOffset.UtcNow.ToUnixTimeSeconds();
        var insertResp = await client.InsertMemoryAsync(new InsertMemoryParams
        {
            Title = "test-key-1",
            Content = "The capital of France is Paris.",
            Namespace = ns,
            CreatedAt = nowSeconds,
            UpdatedAt = nowSeconds,
        });
        Assert.True(insertResp.Success, "insert should succeed");

        // Give the backend time to index
        await Task.Delay(TimeSpan.FromSeconds(2));

        // ── Recall ──
        var recallResp = await client.RecallMemoryAsync(new RecallMemoryParams { Namespace = ns });
        Assert.True(recallResp.Success, "recall should succeed");

        // ── Query ──
        var queryResp = await client.QueryMemoryAsync(new QueryMemoryParams
        {
            Query = "What is the capital of France?",
            Namespace = ns,
        });
        Assert.True(queryResp.Success, "query should succeed");

        // ── Delete ──
        var deleteResp = await client.DeleteMemoryAsync(new DeleteMemoryParams { Namespace = ns });
        Assert.True(deleteResp.Success, "delete should succeed");

        // Give the backend time to process deletion
        await Task.Delay(TimeSpan.FromSeconds(1));

        // ── Verify deletion ──
        var verifyResp = await client.RecallMemoryAsync(new RecallMemoryParams { Namespace = ns });
        Assert.True(verifyResp.Success);
    }
}
